### ISSUE TITLE
can't log my accuracy using tf.estimator

### ISSUE THREAD

**Issue by mttbx** [tensorflow/tensorflow#29471]
I can't log during evalution. It seems tf.train.LoggingTensorHook can only log during training. The tf.train.LoggingTensorHook export the prototxt into my terminal when evalutation.
Here's some of my code

```

tf.logging.set_verbosity(tf.logging.INFO)

train_tensors_to_log = {
    "total_loss": "summary/total_loss",
    "box_loss": "summary/box_loss",
    "cls_loss": "summary/cls_loss",
    "reg_loss": "summary/reg_loss"
}
train_logging_hook = tf.train.LoggingTensorHook(tensors=train_tensors_to_log, at_end=True)

eval_tensors_to_log = {
    "total_loss": "summary/total_loss",
    "box_loss": "summary/box_loss",
    "cls_loss": "summary/cls_loss",
    "reg_loss": "summary/reg_loss",
    "mAP": "eval_summary/mAP"
}
eval_logging_hook = tf.train.LoggingTensorHook(tensors=eval_tensors_to_log, every_n_iter=10)

estimator = tf.estimator.Estimator(model_fn, config=run_config)

train_spec = tf.estimator.TrainSpec(
    input_fn = train_input_fn,
    max_steps = 100000,
    hooks=[train_logging_hook]
)
eval_spec = tf.estimator.EvalSpec(
    input_fn = eval_input_fn,
    steps = 100,
    hooks=[eval_logging_hook],
    throttle_secs=60
)
tf.estimator.train_and_evaluate(
    estimator,
    train_spec,
    eval_spec
)

```

Here's my terminal export:
```

INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x153\x88\xf0?', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15\t;A@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15\xab \xa2@'
INFO:tensorflow:Evaluation [10/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15B\xa5\x08@', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15M\x14r@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15\xe2\xbd\xc2@' (2.541 sec)
INFO:tensorflow:Evaluation [20/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15z \x0c@', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15\x1b;<@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15\xe4\x8e\xa9@' (1.154 sec)
INFO:tensorflow:Evaluation [30/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15\xf4w\x00@', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15\\\xca+@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15B\x82\x9b@' (1.198 sec)
INFO:tensorflow:Evaluation [40/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15\xa2\xcf\x0f@', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15wt{@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15&\x03\xcb@' (1.140 sec)
INFO:tensorflow:Evaluation [50/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15 \x85\x02@', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15s\x9c\x1e@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15\xe4\xf1\x95@' (1.120 sec)
INFO:tensorflow:Evaluation [60/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15\x11\x80\xf7?', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15\xfa3[@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15\x1b\xdb\xb0@' (1.125 sec)
INFO:tensorflow:Evaluation [70/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15\xd9>\xf2?', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15fk\x1e@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15\x83&\x91@' (1.167 sec)
INFO:tensorflow:Evaluation [80/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15\xc4\xe1\xcf?', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15\x00^\x08@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15\x15\x11{@' (1.152 sec)
INFO:tensorflow:Evaluation [90/100]
INFO:tensorflow:box_loss = b'\n\x17\n\x10summary/box_loss\x15\xb4\x8d\x03@', cls_loss = b'\n\x17\n\x10summary/cls_loss\x15s\x15Y@', mAP = b'\n\x17\n\x10eval_summary/mAP\x15\x00\x00\x00\x00', reg_loss = b'\n\x17\n\x10summary/reg_loss\x157#,>', total_loss = b'\n\x19\n\x12summary/total_loss\x15\xae\xb2\xb3@' (1.158 sec)
INFO:tensorflow:Evaluation [100/100]
INFO:tensorflow:Finished evaluation at 2019-06-06-02:27:09
INFO:tensorflow:Saving dict for global step 3000: global_step = 3000, loss = 5.306427

```

**Comment by gadagashwini** [comment]
@mttbx Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?

Make sure you also include the exact command if possible to produce the output included in your test case. If you are unclear what to include see the issue template displayed in [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).

We ask for this in the issue submission template, because it is really difficult to help without that information. Thanks!

**Comment by mttbx** [comment]
Have I written custom code: yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.13
- Python version: 3.5
- CUDA/cuDNN version: 10.0 / 7.5
- GPU model and memory: 7.5, 24GB
- exact command to reproduce: not know yet, but I've show my code to you.

**Comment by gadagashwini** [comment]
@mttbx Code snippet looks incomplete to reproduce the issue. Please provide the complete code to reproduce. Thanks!

**Comment by mttbx** [comment]
This should produce the result, as matter of fact, this issue should be produced with any dataset and any net. 
```
import sys
import os
import pdb
import tensorflow as tf
import numpy as np

from dataset import dataset
from basemodel import net

batch_size = 4
imsz = [512, 512]


def train_input_fn():
    dataset = dataset(batch_size, imsz, './dataset/train.tfrecord')
    return dataset.get()

def eval_input_fn():
    dataset = dataset(batch_size, imsz, './dataset/test.tfrecord')
    return dataset.get()


def model_fn(features, labels, mode):
    is_training = mode == tf.estimator.ModeKeys.TRAIN
    imgs = features
    gts = labels
    
    model = net(batch_size, imsz, is_training)
    model.predict(imgs)
    total_loss, subloss = \
        model.loss(gts)
    

    with tf.name_scope('summary'):
        tf.summary.scalar('subloss', subloss)
        tf.summary.scalar('total_loss', total_loss)


    train_op=None
    if is_training:
        with tf.name_scope('train'):
            global_step = tf.train.get_or_create_global_step()
            trainable_variables = tf.trainable_variables()

            optimizer = tf.train.AdamOptimizer(
                learning_rate=0.0001,
                beta1=0.9,
                beta2=0.999,
                epsilon=1e-08
            )
            
            up_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
            with tf.control_dependencies(up_ops):
                train_op = optimizer.minimize(total_loss, global_step = global_step)

    return tf.estimator.EstimatorSpec(
        mode=mode,
        loss=total_loss,
        train_op=train_op)
    )

if __name__ == '__main__':

    run_config = tf.estimator.RunConfig(
        model_dir='./model',
        tf_random_seed=None,
        save_summary_steps=100,
        save_checkpoints_steps=100,
        session_config=None,
        keep_checkpoint_max=5,
        keep_checkpoint_every_n_hours=1,
    )


    tf.logging.set_verbosity(tf.logging.INFO)

    train_tensors_to_log = {
        "subloss": "summary/subloss",
        "total_loss": "summary/total_loss",
    }
    train_logging_hook = tf.train.LoggingTensorHook(tensors=train_tensors_to_log, at_end=True)

    eval_tensors_to_log = {
        "subloss": "summary/subloss",
        "total_loss": "summary/total_loss",
    }
    eval_logging_hook = tf.train.LoggingTensorHook(tensors=eval_tensors_to_log, every_n_iter=1)


    estimator = tf.estimator.Estimator(model_fn, config=run_config)

    train_spec = tf.estimator.TrainSpec(
        input_fn = eval_input_fn,
        max_steps = 1000000000,
        hooks=[train_logging_hook]
    )
    eval_spec = tf.estimator.EvalSpec(
        input_fn = eval_input_fn,
        steps = 10,
        hooks=[eval_logging_hook],
        throttle_secs=1
    )
    tf.estimator.train_and_evaluate(
        estimator,
        train_spec,
        eval_spec
    )

            
            
            
            
            
            
            
        
```

**Comment by jvishnuvardhan** [comment]
@mttbx Please provide a standalone code to reproduce the issue. You could provide a GitHub gist or any other way to share simple standalone code. Current code cannot run as it is due to import issues other errors. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by yaronhe** [comment]
The same problem!  Has anyone solved this？

**Comment by jvishnuvardhan** [comment]
@yaronhe Please create a new issue with a standalone code to reproduce the issue? Thanks!
